### ISSUE TITLE
Cluster Config UI fails when all nodes have labels

### ISSUE THREAD

**Issue by thatmidwesterncoder** [rancher/dashboard#9888]
**Setup**
- Rancher version: 2.7/2.8, tested back on 2.7.0
- Rancher UI Extensions: n/a
- Browser type & version: chrome/firefox, tested both

**Describe the bug**

When creating a cluster via the UI/terraform/etc if _all_ of the machines have labelSelectors on them the UI fails to load and just crashes.

**To Reproduce**

Apply this terraform: 
<details><summary>Terraform block</summary>
<p>

```hcl
terraform {
  required_providers {
    rancher2 = {
      source  = "terraform.local/local/rancher2"
      version = "3.2.0-rc5"
    }
  }
}

provider "rancher2" {
  api_url   = "<REDACTED>"
  token_key = "<REDACTED>"
  insecure  = true
}

resource "rancher2_cloud_credential" "rancher2_cloud_credential" {
  name = "tf-creds-k3s"
  amazonec2_credential_config {
    access_key = "<REDACTED>"
    secret_key = "<REDACTED>"
  }
}

resource "rancher2_machine_config_v2" "rancher2_machine_config_v2" {
  generate_name = "tf-rke2"
  amazonec2_config {
    ami            = ""
    region         = "<REDACTED>"
    security_group = ["<REDACTED>"]
    subnet_id      = "<REDACTED>"
    vpc_id         = "<REDACTED>"
    zone           = "<REDACTED>"
    root_size      = 50
  }
}

resource "rancher2_cluster_v2" "rancher2_cluster_v2" {
  name                                     = "jkeslar-validate13"
  kubernetes_version                       = "v1.26.8+rke2r1"
  enable_network_policy                    = false
  default_cluster_role_for_project_members = "user"
  rke_config {
    machine_selector_config {
      machine_label_selector{
        match_labels = {
          "key" = "value"
        }
      }
    }
    machine_pools {
      name                         = "pool1"
      machine_labels = {
        "key" = "value" 
      }
      cloud_credential_secret_name = rancher2_cloud_credential.rancher2_cloud_credential.id
      control_plane_role           = true
      etcd_role                    = true
      worker_role                  = true
      quantity                     = 1
      machine_config {
        kind = rancher2_machine_config_v2.rancher2_machine_config_v2.kind
        name = rancher2_machine_config_v2.rancher2_machine_config_v2.name
      }
    }
  }
}
```

</p>
</details> 

Alternatively, create a cluster in the UI and make sure the machineSelectorConfig yaml looks like this: 
```yaml
    machineSelectorConfig:
      - config:
        machineLabelSelector:
          matchExpressions:
            - key: string
              operator: string
              values:
                - string
          matchLabels:  
            key: string
```

Create the cluster, wait for it to become active, go to the UI, and click the `Config` button.

**Result**
Errors in console, fails to load the page. 

**Expected Result**

The cluster config should load. 

**Screenshots**

![image](https://github.com/rancher/dashboard/assets/135059506/61c60e1b-150f-4d1a-9dba-7510b1cadf9e)
![image](https://github.com/rancher/dashboard/assets/135059506/a3086a08-b466-45b5-9041-84480317a6ca)

**Additional context**

This was found initially during QA testing some terraform changes - the UI fails to handle this specific case where every node has a label selector on it.

The piece of failing code is here: https://github.com/rancher/dashboard/blob/0076a24eff9f489c96c85b14e35d40461dd0a2dd/shell/models/provisioning.cattle.io.cluster.js#L815-L819

It fails on the last function call since no agent node was found without a label.

**Comment by gaktive** [comment]
Possible connection with SURE-7011 (internal reference) whereby S3 backup configuration produces the error `Cannot read properties of undefined (reading ‘length’)` though some deciphering of that Terraform setup is still required.

**Comment by gaktive** [comment]
Another internal reference: SURE-7212

**Comment by torchiaf** [comment]
@thatmidwesterncoder  users can add configuration entries in `agentConfig` field, in RKE2 clusters creation, i.e. `protect-kernel-defaults`; you can find it as a checkbox at the bottom of `Advanced` tab (`Raise errors if ...`):

![image](https://github.com/rancher/dashboard/assets/26394656/b1643bc7-f75f-4532-bffb-39fa87fb9a60)


Where those entries should be saved when every node has a label selector ( --> `agentConfig` is undefined) ?
As per documentation, it seems that they could be saved in `machineGlobalConfig` section:
https://ranchermanager.docs.rancher.com/reference-guides/cluster-configuration/rancher-server-configuration/rke2-cluster-configuration#machineselectorconfig

**Comment by izaac** [comment]
This is tightly related to other issue. Both could be validated during the same manual validation using terraform and manual RKE2 configuration. cc @yonasberhe23 

Related issue comment: https://github.com/rancher/dashboard/issues/10045#issuecomment-1973432313

**Comment by aalves08** [comment]
@gaktive @nwmac this is also linked to SURE-6770... Would SURE-6770 be closed as well?

**Comment by izaac** [comment]
@aalves08 this is on Isabela's to-do items, she's getting issues validated according to priorities so she would take a look as soon as possible. cc @IsaSih

**Comment by aalves08** [comment]
@izaac not a priority. I just wanted to double-check since I believe the zube bot has been messing with the issue labels :P

**Comment by gaktive** [comment]
@aalves08 once QA has marked this as Done, then yes SURE-6770 & SURE-7212 can be closed.

**Comment by IsaSih** [comment]
I manually tested this issue and the current behavior corresponds to what is expected.

**Comment by IsaSih** [comment]
Taking a look at each of the JIRA tickets mentioned in the comments, I found that SURE-6770 is a different test case, where this property is not present at all. 

```
machineSelectorConfig:
  - config: null
```

**Comment by torchiaf** [comment]
@IsaSih What is the status of this one?

**Comment by nwmac** [comment]
@IsaSih ping - any update - how can eng help unblock this for you?

**Comment by IsaSih** [comment]
Tests pass on v2.9-3ae1c6ac8d3524f8ded4544f1db75fc9e07911a5-head


![Image](https://github.com/rancher/dashboard/assets/70187881/ca603f3f-2acd-40d8-b1ee-315d663304cc)
